File: frontend/src/app/routes/netex-route-variant.ts
import ChbQuay from '@quays/chb-quay'
import NetexRouteQuay from '@quays/netex-route-quay'

export default class NetexRouteVariant {
    public id!: number;
    public name!: string;
    public lineNumber!: string;
    public lineRef!: string;
    public directionType!: string;
    public quays: NetexRouteQuay[] = [];
        
    public get fromQuay(): NetexRouteQuay {
        return this.quays[0];
    }

    public get toQuay(): NetexRouteQuay {
        return this.quays[this.quays.length - 1];
    }

}

File: src/main/java/nl/haltedata/netex/dto/NetexRouteVariant.java
package nl.haltedata.netex.dto;

import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.NamedAttributeNode;
import jakarta.persistence.NamedEntityGraph;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import nl.haltedata.chb.dto.ChbQuay;

@Entity
@Table(schema="netex")
@Getter
@Setter
@EqualsAndHashCode(of = "id")
@NamedEntityGraph(
        name = "netexRouteVariant",
        attributeNodes = {
            @NamedAttributeNode("netexLine"),
            @NamedAttributeNode("quays"),
        })
public class NetexRouteVariant {
    @Id
    private Long id;
    private String lineNumber;
    private String colour;
    private Long quayCount;
    private String directionType;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "line_ref")
    private NetexLine netexLine;
    @OneToMany(mappedBy = "routeVariant", fetch = FetchType.LAZY)
    private List<NetexRoute> netexRoutes;
    @OneToMany(mappedBy = "routeVariant")
    private List<NetexRouteVariantQuay> quays;
}
